src/django_components/component_media.py

```diff
@@ -41,6 +41,21 @@
 COMP_MEDIA_LAZY_ATTRS = ("media", "template", "template_file", "js", "js_file", "css", "css_file")
 
 
+# Sentinel value to indicate that a media attribute is not set.
+# We use this to differntiate between setting template to `None` and not setting it at all.
+# If not set, we will use the template from the parent component.
+# If set to `None`, then this compoenent has no template.
+class Unset:
+    def __bool__(self) -> bool:
+        return False
+
+    def __repr__(self) -> str:
+        return "<UNSET>"
+
+
+UNSET = Unset()
```

Comment:
Since `None` is now a valid value for the component media properties, I used a sentinel value `UNSET` to differentiate between explicit `None` and unset fields.